### PR TITLE
Support block delete with word and line modifiers

### LINKED ISSUE #57015: [macOS] Textfields should support CMD (⌘) + Backspace and Control + Backspace
## Use case
Normally macOS-Apps (or desktop apps in generell) support CMD (⌘) + Backspace and Control + Backspace in Textfields.

## Proposal
CMD (⌘) + Backspace: Delete Line
Control + Backspace: Delete word left to the cursor

MacOSTextfieldKeys

### LINKED ISSUE #77547: Delete by word does not work
Control/Alt + Delete/Backspace should delete entire word, but it is currently not implemented.

### PR DESCRIPTION
**Pull request by canastro**
This PR focuses on supporting modifiers to delete blocks of text in a editable widget.

### Shortcuts per OS
* macOS
  * **CMD (⌘) + Backspace**: Delete all words left to the cursor
  * **Option + Backspace**: Delete word left to the cursor
* Non macOS
  * **Alt + Backspace**: Delete all words left to the cursor
  * **Ctrl + Backspace**: Delete word left to the cursor

### Some cases for reference:

**case: 'text with| multiple words'**
word modifier + backspace: 'text | multiple words'
line modifier + backspace: '| multiple words'
word modifier + delete: 'text with| words'
line modifier + delete: 'text with|'

**case: 'text with multiple\n|words'**
word modifier + backspace: 'text with |words'
line modifier + backspace: 'text with multiple\n|words'
word modifier + delete: 'text with multiple\n|'
line modifier + delete: 'text with multiple\n|'

**case: 'text with multiple|\nwords'**
word modifier + backspace: 'text with |\nwords'
line modifier + backspace: '|\nwords'
word modifier + delete: 'text with multiple|'
line modifier + delete: 'text with multiple|\nwords'

**case: 'text with\n|\nmultiple words' in obscure text**
word modifier + backspace: '|\nmultiple words'
line modifier + backspace: '|\nmultiple words'
word modifier + delete: 'text with\n|'
line modifier + delete: 'text with\n|'

**case: 用多個塊測試** (cursor at offset 4)
backspace: '用多個測試' (cursor at offset 3)
word modifier + backspace: '用多個測試' (cursor at offset 3)
line modifier + backspace: '測試' (cursor at offset 0)
word modifier + delete: '用多個塊試' (cursor at offset 4)
line modifier + delete: '用多個塊' (cursor at offset 4)

**case: برنامج أهلا بالعالم** (cursor at offset 4 - from the right)
backspace: برنمج أهلا بالعالم  (cursor at offset 3 - from the right)
word modifier + backspace: مج أهلا بالعالم (cursor at offset 0 - from the right)
word modifier + delete: برنا أهلا بالعالم  (cursor at offset 4 - from the right)

### Related issue: 
* Fixes https://github.com/flutter/flutter/issues/57015
* Fixes https://github.com/flutter/flutter/issues/77547

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by canastro** [comment]
@LongCatIsLooong @gspencergoog 

This PR aims to fix the issue that was introduced on https://github.com/flutter/flutter/pull/77172

Here is the commit that fixes: https://github.com/flutter/flutter/pull/79695/commits/b1d14c2ad5388919e796f9aa4a7bd504241d561e 
This was probably a test that was updated between my last rebase and the PR being merged.

Sorry for the inconvenient.

**Review by justinmc — APPROVED** [review]
LGTM 👍 

Ok looks like it was just a test added that used the keyboard.  Easy fix.

**Comment by justinmc** [comment]
Google testing wasn't working for me so I've just set it to green.  I think we should be ok.

**Comment by justinmc** [comment]
I'm holding off on merging this because of a crash reported in RenderEditable.  It's not caused by this PR, but I don't want to complicate things.  Will merge when that's resolved.

**Comment by justinmc** [comment]
Everything is clear with the other bug so I'm merging this!

**Comment by justinmc** [comment]
Thanks for all of your work on this @canastro.

**Pull request merged**
